notebook/static/services/kernels/kernel.js

```diff
@@ -882,8 +882,14 @@ define([
         var that = this;
         this._msg_queue = this._msg_queue.then(function() {
             return serialize.deserialize(e.data);
-        }).then(function(msg) {return that._finish_ws_message(msg);})
-        .catch(utils.reject("Couldn't process kernel message", true));
+        }).then(function(msg) {
+            try {
+                return that._finish_ws_message(msg);
+            } catch (e) {
+                console.error("Couldn't process kernel message", e);
+                return Promise.resolve();
```

Comment:
So the difference here is that you want to promise chain to continue, even if there is a failure?

If so, you can continue to use catch, just don't use the utils.reject function.